src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1971,10 +1971,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
     }
 
     private def ensurePredefParentsAreInSameSourceFile(template: Template) = {
-      val parentSyms = template.parents map (_.symbol) filterNot (_ == AnyRefClass)
       val PredefModuleFile = PredefModule.associatedFile
-      if (parentSyms exists (_.associatedFile != PredefModuleFile))
-        context.error(template.pos, s"All parents of Predef must be defined in ${PredefModuleFile}.")
+      if (template.parents.exists(p => p.symbol != AnyClass && p.symbol.associatedFile != PredefModuleFile))
+        context.error(template.pos, s"All parents of Predef must be defined in $PredefModuleFile.")
```

Comment:
Noting switch to AnyClass.

Reply:
Surely you mean `AnyRefClass`

Reply:
I think it was `filterNot(_ == AnyRefClass)` and now `exists(_.symbol != AnyClass)`. I'm not sure, without waking up completely, if it matters.

Reply:
It was `map(_.symbol).filterNot(_ == AnyRefClass)` 🤷‍♂️